app/controllers/solutions_controller.rb

```diff
@@ -0,0 +1,19 @@
+class SolutionsController < ApplicationController
+  def create
+    @question = Question.find(params[:question_id])
+    @solution = @question.solutions.build(solution_params)
+    @solution.question_id = @question.id
+    if @solution.save
+      redirect_to question_path, notice: "回答を投稿しました。"
```

Comment:
この書き方では次のエラーが出ます。

> No route matches {:action=>"show", :controller=>"questions", :question_id=>"17"}, missing required keys: [:id]

pathにidを渡していないのが原因ですね。

```suggestion
      redirect_to @question, notice: "回答を投稿しました。"
```

Reply:
pathにidを渡せるように修正しました。